salt/utils/data.py

```diff
@@ -67,17 +67,16 @@ def compare_lists(old=None, new=None):
     return ret
 
 
-@jinja_filter('json_decode_dict')
 def decode_dict(data):
     '''
-    JSON decodes as unicode, Jinja needs bytes...
+    Decode all values to Unicode
     '''
     rv = {}
     for key, value in six.iteritems(data):
-        if isinstance(key, six.text_type) and six.PY2:
-            key = key.encode('utf-8')
-        if isinstance(value, six.text_type) and six.PY2:
-            value = value.encode('utf-8')
+        if six.PY2 and isinstance(key, str):
+            key = key.decode(__salt_system_encoding__)
```

Comment:
Assuming I am reading the code correctly. Needs to remain as UTF-8.  Inside Salt should be Unicode Vars or UTF-8. Only at the final output (edges) should be be converted into something else.

The filter should be called e.g. `text_to_bytes(text_type='acsii')` (defaulting to UTF-8), because people mix up encode/decode even if your a python programmer.

Reply:
If you're on a unicode locale, `__salt_system_encoding__` _is_ "utf-8". Also, I don't think you're reading the code correctly. This is a new function which recursively decodes strings, and it's not used in Salt to my knowledge, so I'm not sure what the concern is.